### ISSUE TITLE
EF_STAFF_HOLY does not do what its description says

### ISSUE THREAD

**Issue by angband-import** [angband/angband#3736]
**Reported by myshkin on 19 Aug 2012 04:05 UTC**
Description: "inflicts damage on evil creatures you can see, cures 50 hit points, heals all temporary effects and grants you protection from evil"

(As srd_ notes, we should probably swap "cures" and "heals" here regardless.)

Code: STAFF_HOLY only clears TMD_POISONED, TMD_AFRAID, TMD_STUN, and TMD_CUT. It does not affect TMD_SLOW, TMD_BLIND, TMD_PARALYZED, TMD_CONFUSED, or TMD_IMAGE; of course, TMD_PARALYZED doesn't make sense here.

How should we fix this?

**Comment by angband-import** [comment]
**Comment by myshkin on 19 Aug 2012 04:17 UTC**
From IRC:

```
Derakon: Myshkin: make it a super-Staff of Curing, IMO.
Derakon: So it should clear all bad timered effects.
```

**Comment by angband-import** [comment]
**Comment by myshkin on 19 Aug 2012 04:48 UTC**
From IRC: 
```
d_m: myshkin: re 1670, i agree with Derakon
d_m: given that it currently is pretty bad, i'd say make it good. if it's too good we'll hear about it
```

**Comment by angband-import** [comment]
**Comment by magnate on 19 Aug 2012 12:15 UTC**
Agreed - let's have it clear confusion, hallucination, slowing and blindness too.

**Comment by angband-import** [comment]
**Comment by myshkin on 19 Aug 2012 19:57 UTC**
My initial list omitted TMD_TERROR and TMD_AMNESIA. I should also note that more generally we have a whole host of effects that clear some subset of negative status effects, and most of them do something slightly different. effects.c could use some refactoring. I think that's more for 3.5, though.

**Comment by angband-import** [comment]
**Comment by pnd10 on 25 Aug 2012 08:23 UTC**
Fixed in c2abc5f by myshkin
